src/rng.cpp

```diff
@@ -0,0 +1,53 @@
+/*
+ * rng.cpp
+ *
+ *  Created on: 25 Mar 2020
+ *      Author: gc1610
+ */
+
+#include <Rcpp.h>
+
+float rgenexp(float av) {
+	return Rcpp::rexp(1, av)[0];
```

Comment:
It might not make any difference and probably worth checking with Rich or someone who knows more about this than me, but I think the Rcpp:: RNGs are vectorised and scalar versions are accessed in Rcpp via the `R` namespace. See: https://gallery.rcpp.org/articles/random-number-generation/ for more info.
Just wondering if this will have any considerable performance impact?

Reply:
Good spot! It makes the code a bit nicer